### DISCUSSION TITLE
Error when installing/using parser for Dockerfile

### DISCUSSION THREAD

**Discussion by Masber** [nvim-treesitter/nvim-treesitter#6663] [Q&A]
Dear treesitter community,

I just installed nvim v0.10 and I am getting the following error when opening telescope

```
Error executing vim.schedule lua callback: ...local/share/nvim/runtime/lua/vim/treesitter/language.lua:112: Failed to load parser for language 'dockerfile': uv_dlopen: no error
stack traceback:
        [C]: in function '_ts_add_language'
        ...local/share/nvim/runtime/lua/vim/treesitter/language.lua:112: in function 'add'
        ...l/share/nvim/runtime/lua/vim/treesitter/languagetree.lua:111: in function 'new'
        /usr/local/share/nvim/runtime/lua/vim/treesitter.lua:41: in function '_create_parser'
        /usr/local/share/nvim/runtime/lua/vim/treesitter.lua:108: in function 'get_parser'
        ...m/lazy/telescope.nvim/lua/telescope/previewers/utils.lua:141: in function 'ts_highlighter'
        ...m/lazy/telescope.nvim/lua/telescope/previewers/utils.lua:119: in function 'highlighter'
        ...scope.nvim/lua/telescope/previewers/buffer_previewer.lua:248: in function ''
        vim/_editor.lua: in function <vim/_editor.lua:0>
```

I am assuming the parser name for Dockerfile is incorrect, this is my treesitter configuration

```
  {
    'nvim-treesitter/nvim-treesitter',
    build = ':TSUpdate',
    config = function()
      require('nvim-treesitter.configs').setup {
        ensure_installed = { 'bash', 'c', 'html', 'lua', 'markdown', 'vim', 'vimdoc', 'go', 'rust', 'dockerfile' },
        auto_install = true,
        highlight = { enable = true },
        indent = { enable = true },
      }
    end,
  },
```

I am assumming the parser name is not dockerfile but something different, I went to the parser github page but it does not say which name should I use.

https://github.com/camdencheek/tree-sitter-dockerfile

Any idea on what is causing this error or how to solve it?

thank you very much

**Comment by ashemedai** [comment]
The main README.md lists the name as `dockerfile`. And in fact, I do have it in my Neovim 0.10.0 setup:

```
ls ~/.local/share/nvim/lazy/nvim-treesitter/parser/dockerfile.so
/home/asmodai/.local/share/nvim/lazy/nvim-treesitter/parser/dockerfile.so
```

That's also what you see with a `:TSUpdate dockerfile`: `[nvim-treesitter] [1/1] Treesitter parser for dockerfile has been installed` after removing it from your environment.

And that just fixed it for me at least. So you might want to delete the shared object for dockerfile, if you have it, and reinstall it. I have no idea why it broke when Neovim 0.10.0 was updating all the treesitter parsers though.

**Comment by jaswdr** [comment]
What helped in my case is adding the "dockerfile" in the list of "ensure_installed", like bellow:

```
{
  "nvim-treesitter/nvim-treesitter",
  opts = {
    ensure_installed = {
      ...
      "dockerfile",
      ...
    },
  },
},
```
